### DISCUSSION TITLE
Trouble installing Underpin

### DISCUSSION THREAD

**Discussion by Niloys7** [alexstandiford/underpin#46] [Q&A] [answered]
Hello,
Thanks for creating such a beautiful framework. I would like to use it for my upcoming WordPress projects but I found some issues where I was stuck.
Example: I tried to install the "Underpin Custom Post Type Loader" ext for my plugin but it's give me a fatal error when I tried to the readme.MD introductions.  when I check the core "Underpin" plugin I found that the latest version is not supported the "underpin()" function. but the ext's readme.md files are not updated or do not have details for how it's working with the latest version.

I found some issues too when I tried to load packages via composer.
Example: Unerpin doesn't have a version declared on composer but "Underpin Custom Post Type Loader" ext have set version 2.0 for Underpin https://i.imgur.com/FtlpsTw.png :(  

- Niloy

**Comment by alexstandiford** [comment, marked as answer]
Hi @Niloys7, thanks! I'm a fan :smile:. I went ahead and converted this to a conversation. There's definitely some action-items here, but I think it's best we start with a discussion and turn action items into issues to get these things fixed.

It seems like the wrong version of the Underpin Custom Post Type loader got installed in your setup.

Can you tell me a little about the steps you took to set up Underpin? There's many ways to do this, and it's possible that something is mis-configured in the system to allow it to work properly in the way you're trying to-do it. 

The `underpin()` function [was removed in 2.0](https://github.com/Underpin-WP/underpin#underpin-function-removed) because there's just no need for Underpin to have its own instance. Instead, you'll need to create your own plugin instance of Underpin, like how it's done in the [plugin boilerplate](https://github.com/Underpin-WP/underpin-plugin-boilerplate). This function serves as the entry-point for your plugin in its entirety. Anywhere where you see `underpin()` in the docs, for now assume that you should instead use whatever [your declared function that instantiates your plugin's instance is called](https://github.com/Underpin-WP/underpin-plugin-boilerplate/blob/cd68e0443daec8d627dcf61ec5c15c7ba9202589/bootstrap.php#L30-L38).

Have you checked out the [mu plugin boilerplate](https://github.com/Underpin-WP/mu-plugin-boilerplate)? That might help get you in the right direction with how I set this up to get it going. You don't _have_ to install Underpin as a mu-plugin, I just usually opt for this.

I'm absolutely open to suggestions on how to improve this experience, and if I'm missing something in my `composer.json` file that can make this more-smooth, feel free to submit a PR!

**I really appreciate your feedback!**

**Reply by Niloys7** [reply]
Thanks @alexstandiford  . Changing `underpin()` to my custom function name is works.

**Comment by Niloys7** [comment]
the `underpin()` function is still there :) `Release 2.1` [https://github.com/Underpin-WP/logger-loader](https://github.com/Underpin-WP/logger-loader)

 ![fatal error on logger-loader ext](https://i.imgur.com/XLIKNsv.png)

**Reply by alexstandiford** [reply]
What is installing the logger loader? That was added internally and the loader is only for versions older than 2.0

**Reply by Niloys7** [reply]
can't remember right now, :(  yesterday I tried to test all extensions.

**Comment by Niloys7** [comment]
Here is the Plugin [cixpin.zip](https://github.com/Underpin-WP/underpin/files/7673612/cixpin.zip)
please check and let me know, why I am getting errors.
In the `cixpin.php` file only `custom_post_types` is working, I also added `roles,admin_notices, options` but these appear with a fatal error
